### ISSUE TITLE
udevadm trigger is needed before ofonod works

### ISSUE THREAD

**Issue by pavelmachek** [postmarketOS/pmbootstrap#700]
Current init scripts do not do udevadm trigger, so ofonod starts and immediately exits. Running udevadm trigger fixes that.

```
localhost:~$ sudo udevadm trigger
localhost:~$ ps aux | grep ofonod
 1639 user       0:00 grep ofonod
localhost:~$ sudo ofonod
localhost:~$ ps aux | grep ofonod
 1642 root       0:00 ofonod
 1644 user       0:00 grep ofonod
localhost:~$ 

```

**Comment by ollieparanoid** [comment]
There's an OpenRC service, that explicitly triggers udev (in runlevel `default`). Maybe that only gets executed after `ofonod` starts (i.e. we need to put `ofono` into another level?)

https://github.com/alpinelinux/aports/blob/2fbea8b8dae00613de97862e509c6f76988ddeaf/main/eudev/setup-udev#L24

`ofono`'s current runlevel is also `default`, you could try to change that here:
https://github.com/postmarketOS/pmbootstrap/blob/71a244fca656896d1dbfb3e3115fe160200a7ed2/aports/device/device-nokia-rx51/device-nokia-rx51.post-install#L8-L15

**Comment by craftyguy** [comment]
Hmm, I had the trigger in there ([see here](https://github.com/postmarketOS/pmbootstrap/commit/26204891a8038ba18a1eafded1b85da3e8878f10#diff-ffe196c1d91443d8ecd7d06bfa315884L150) but removed it when that udev commit was submitted. I will re-add the udev-trigger service to the default runlevel since that seems to have resolved this issue.

**Comment by ollieparanoid** [comment]
Oh, I just realize that the [`udev-setup`](https://github.com/alpinelinux/aports/blob/2fbea8b8dae00613de97862e509c6f76988ddeaf/main/eudev/setup-udev#L21-L26) scripts on *starts* the services, it does not *enable* them (and it doesn't even start them when called in postmarketos-base' post-install script, because we do it as dry run, which is intended)!

```shell
	if [ -z "$dryrun" ] && ! rc-service --quiet udev status; then
		rc-service udev start
		rc-service udev-postmount start
		rc-service udev-trigger start
		rc-service udev-settle start
fi
```

**Comment by pavelmachek** [comment]
I tried adding 9900c5e commit, but it did not seem to help. Should "need udev-settle" be added to /etc/init.d/ofono ?

**Comment by ollieparanoid** [comment]
Yeah, I think depending on `udev-settle` is the right way (not sure about the OpenRC specific syntax in the service files though)!

After staring at the `udev-setup` code for a bit, I realized now, that it *does* enable the services - it just doesn't use the `rc-update` script from OpenRC, but create the symlinks manually.

So all it does, is in fact:
* disable services: hwdrivers and mdev
* add services (to runlevel `sysinit`): udev-trigger udev udev-postmount

(It might be, that `udev-postmount` is not needed, see [Gentoo's post-install hook](https://gitweb.gentoo.org/repo/gentoo.git/tree/sys-fs/udev-init-scripts/udev-init-scripts-32.ebuild).)

So for postmarketOS, this means that I would not call `setup-udev` anymore, but enable the two or three services manually. That makes it way easier to understand.

**Comment by pavelmachek** [comment]
Still happens in recent version ( b3351ab3059c643cb4a5b3830c0ebdd1bbac2436 ). I tried checking out fix/700_rx51_udev_trigger , but it did not build. Tried removing the build directory, but that made build fail different way :-(

**Comment by craftyguy** [comment]
it's probably not building because I needed to rebase that branch. I just did the rebase, can you give the fix/700_rx51_udev_trigger branch another try?

**Comment by pavelmachek** [comment]
Thanks a lot. I updated to 03f3833...I re-ran install and tested, but no change; still needed to do udevadm trigger before ofonod would start.

**Comment by ollieparanoid** [comment]
Thanks for rebasing @craftyguy and for testing @pavelmachek! How do we move forward with this then, is there any way I can help?

**Comment by ollieparanoid** [comment]
So in other words, the open PR did not fix the issue. Maybe someone with an N900 can look into this more (just play with ofono on the N900, and possibly try to fix this issue)? @pavelmachek wrote elsewhere that he won't have time for this any time soon.

**Comment by scintill** [comment]
I did not have this issue on my Samsung phone; it works from standard boot. Any more information on why ofonod exits for you? Is the OFONO_DRIVER property missing before your manual trigger? You can check with a command like `udevadm info -q all /path/to/device`

It seems that both the ofono package and pmOS's n900 device package have similar udev rules files for n900. Do they conflict in some way?
